swupd/packs_test.go

```diff
@@ -255,6 +255,64 @@ func TestCreatePackZeroPacks(t *testing.T) {
 	mustValidateZeroPack(t, ts.path("www/20/Manifest.shells"), ts.path("www/20/pack-shells-from-0.tar"))
 }
 
+func TestCreatePackNonConsecutiveDeltas(t *testing.T) {
+	ts := newTestSwupd(t, "create-pack-ncd")
+	defer ts.cleanup()
+
+	ts.Bundles = []string{"os-core", "contents"}
+
+	contents := strings.Repeat("large", 1000)
+	if len(contents) < minimumSizeToMakeDeltaInBytes {
+		t.Fatal("test content size is invalid")
+	}
+
+	ts.addFile(10, "contents", "/A", contents+"A")
+	ts.addFile(10, "contents", "/B", contents+"B")
+	ts.addFile(10, "contents", "/C", contents+"C")
```

Comment:
Optional: use separate contentsA, contentsB and contentsC (with different contents) to reduce the chance in the future we get deltas other the ones we expect if the delta picking logic change drastically.

Reply:
I'm going to call this good for now, since the test explicitly checks for the deltas it expects. If we change the delta picking logic drastically this test will fail and the developer can update it.